### PR TITLE
Adds total_workers_count, workers_running_a_job and latency to values

### PR DESCRIPTION
**Pull request by stokwell**

### CODE REVIEW

File: lib/telegraf/sidekiq.rb [hunk 1]
```diff
@@ -56,7 +56,10 @@ def call(worker, job, queue)
         }
 
         values = {
-          retry_count: job['retry_count']
+          retry_count: job['retry_count'],
+          total_workers_count: ::Sidekiq.options[:concurrency],
+          workers_running_a_job: ::Sidekiq::Workers.new.size,
+          latency: ::Sidekiq::Queue.new.latency.to_f
```
Comment (franzliedke) [inline lib/telegraf/sidekiq.rb:62]: This only measures latency of the `default` queue. We use different queues in different services / workers.

File: lib/telegraf/sidekiq.rb [hunk 2]
```diff
@@ -56,7 +56,10 @@ def call(worker, job, queue)
         }
 
         values = {
-          retry_count: job['retry_count']
+          retry_count: job['retry_count'],
+          total_workers_count: ::Sidekiq.options[:concurrency],
```
Comment (franzliedke) [inline lib/telegraf/sidekiq.rb:60]: Is this value correct? I would suspect this to be the number of threads running in parallel on the current VM. What if we run multiple processes or one process on multiple VMs, which we both do on production?

**Review by franzliedke — COMMENTED** [review]
High-level feedback: Is this the approach we want? Here, we are determining the new metrics (worker counts, queue latency) every time we start executing a new job.

Some drawbacks I see:
- amount of events differs a lot based on frequency / length ob jobs
- calculation of values actually affects the measured job duration

In addition, the Sidekiq API methods you are calling all execute one or even multiple (N+1) Redis commands. The dangers here are additional load, and an additional source of errors (which are not handled so far).

One totally different approach would be running this regularly via cron or some other form of timer, to have the data at regular intervals, independent from workers being executed, and not affecting them in any other way. Have you considered that variant? What advantages / drawbacks do you see? (IIRC, this variant was used in one of the GitHub repositories you or I linked in chat.)

File: lib/telegraf/sidekiq.rb (hunk 2 above)
Comment (jgraichen) [reply lib/telegraf/sidekiq.rb:60]: Is this value important at all? :thinking:

Comment (stokwell) [reply lib/telegraf/sidekiq.rb:60]: Thanks for the review, I would try to reimplement this metrics https://github.com/yabeda-rb/yabeda-sidekiq and will wrap it in periodical exporter.

File: lib/telegraf/sidekiq.rb (hunk 1 above)
Comment (jgraichen) [reply lib/telegraf/sidekiq.rb:62]: What _exactly_ is `::Sidekiq::Queue.new.latency` measuring? What latency? Redis command latency? The time between job queuing and processing? With or without retries? What about named queues, or multiple queues?

Do we have a definition here, or do we want to implement it ourselves to know what exactly is measured?

Comment (jgraichen) [reply lib/telegraf/sidekiq.rb:62]: The metric point is about a specific job. I would definitely expect the latency to be strictly related to the job only too.

**Pull request closed without merging**